taskcluster/docker/node/build.py

```diff
@@ -75,17 +75,32 @@ def write_package_info(package_info):
 
 def get_buildid():
     now = datetime.utcnow()
-    return now.strftime("%Y%m%d.%H%M%S")
+    # note the `-` (hyphen) in `%-H`
+    # this removes the leading zero
+    # from the hour (leading zeros are not allowed)
+    return now.strftime("%Y%m%d.%-H%M%S")
 
 
 def get_buildid_version(version):
-    """Version schema check+append a `buildid{buildid}` to ensure unique version."""
-    # XXX is there a more precise schema we should verify ourselves against?
-    if len(version.split(".")) not in (1, 2, 3):
-        raise Exception("{version} has too many version parts!")
-    if "buildid" in version:
-        raise Exception("{version} already has a buildid specified!")
-    buildid_version = f"{version}buildid{get_buildid()}"
+    """Check the version's formating and append `date.time` as a buildid to ensure a unique version.
+    The addon's in-tree extension/manifest.json file specifies `major.minor.0`.
```

Comment:
I think we update both JSON files (`package.json` and `manifest.json`) based on the version supplied in the `package.json`, right?

Reply:
Ah good catch